web/src/pages/billing/BillingCostByAnalysis.tsx

```diff
@@ -0,0 +1,253 @@
+            </h1>
+
+            <Grid columns={2} stackable>
+                <Grid.Column width={10}>
+                    {/* There is a Dropdown inside the search Input control to select searchType */}
+                    <Input
+                        fluid
+                        label={
+                            <Dropdown
+                                selection
+                                compact
+                                defaultValue={getDefaultSearchType()}
+                                options={dropdownOptions}
+                                onChange={handleSearchTypeChange}
+                            />
+                        }
+                        labelPosition="right"
+                        placeholder="Search..."
+                        onChange={handleSearchChange}
+                        value={searchTxt}
+                    />
+                </Grid.Column>
+                <Grid.Column width={1}>
+                    <Button compact onClick={() => handleSearchClick()} icon={<SearchIcon />} />
+                </Grid.Column>
+            </Grid>
+
+            <br />
+            <p>
+                {' '}
+                <br />
+                Try these examples:
+                <br />
+                Ar guid: f5a065d2-c51f-46b7-a920-a89b639fc4ba
+                <br />
+                Batch id: 430604, 430605
+            </p>
+        </Card>
+    )
+
+    const gridCard = (gridData: BillingTotalCostRecord[]) => (
+        <Card fluid style={{ padding: '20px', overflowX: 'scroll' }} id="billing-container-data">
+            <HailBatchGrid data={gridData} />
+        </Card>
+    )
+
+    const dataComponent = () => {
+        if (data !== undefined && data.costs.length > 0) {
+            // only render grid if there are available cost data
+            return gridCard(data.costs)
+        }
+
+        // if valid search text and no data return return No data message
+        if (
+            data !== undefined &&
+            searchByType !== undefined &&
+            searchTxt !== undefined &&
+            searchTxt.length > 5
+        ) {
+            return (
+                <p style={{ textAlign: 'center', marginTop: '5px' }}>
+                    <em>No data found.</em>
+                </p>
+            )
+        }
+
+        // otherwise prompt user to search, if not loading already
+        if (!isLoading) {
+            return (
+                <p style={{ textAlign: 'center', marginTop: '5px' }}>
+                    <em>
+                        Enter a search term above, select type and press search button to get
+                        started.
+                    </em>
+                </p>
+            )
+        }
+
+        // otherwise do not render anything
+        return null
+    }
+
+    return (
+        <>
+            {searchCard()}
+            {errorComponent()}
+            {loadingComponent()}
+            {dataComponent()}
```

Comment:
This is totally fine, (I'm not suggesting you change it) as a side note as it gets transpiled to basically the same thing, you can actually do ;)

```
<dataComponent />
```

Reply:
Did not know this one, thanks for pointing it out, kept the original.